src/main/java/com/esri/samples/symbology/symbol_dictionary/SymbolDictionaryController.java

```diff
@@ -74,14 +79,21 @@ private void handleSearchAction() {
     ListenableFuture<List<SymbolStyleSearchResult>> searchResult = dictionarySymbol.searchSymbolsAsync(searchParameters);
     searchResult.addDoneListener(() -> {
       try {
-        List<SymbolStyleSearchResult> symbolResults = searchResult.get();
+        symbolResults = searchResult.get();
 
         Platform.runLater(() -> {
-          // create and add results to listview
-          symbolResults.stream().map(SymbolView::new).collect(Collectors.toCollection(() -> resultList.getItems()));
+          int results = symbolResults.size();
+          if (results > 0) {
+            // create and add results to listview
+            symbolResults.subList(0, Math.min(results, RESULT_SIZE)).stream().map(SymbolView::new).collect(Collectors.toCollection(() -> resultList.getItems()));
```

Comment:
There's an official `Pagination` control for this that you should use: http://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/Pagination.html.